base/menus/ButtonSelectMenu.js

```diff
@@ -0,0 +1,232 @@
+import { get_text_width } from '../utils.js';
+import * as numbers from '../magic_numbers.js';
+import { Window } from '../Window.js';
+
+const FORWARD = 1;
+const BACKWARD = -1;
+const BUTTON_WIDTH = 24;
+const BUTTON_HEIGHT = 24;
+const BUTTON_Y = numbers.GAME_HEIGHT - BUTTON_HEIGHT;
+const TITLE_WINDOW_HEIGHT = BUTTON_HEIGHT - numbers.OUTSIDE_BORDER_WIDTH - numbers.INSIDE_BORDER_WIDTH;
+
+export class ButtonSelectMenu {
+    constructor(game, data, buttons, titles, callbacks, control_manager, title_window_width, dock_right=false) {
+        this.game = game;
+        this.data = data;
+        this.buttons_keys = buttons;
+        this.titles = titles;
+        this.on_cancel = callbacks.on_cancel;
+        this.on_press = callbacks.on_press;
+        this.control_manager = control_manager;
+        this.buttons_number = buttons.length;
+        this.custom_scale = null;
+
+        const max_title_width = get_text_width(this.game, _.maxBy(titles, title => title.length));
+        this.title_window_width = title_window_width !== undefined ? title_window_width : max_title_width + 2 * (numbers.WINDOW_PADDING_H + numbers.INSIDE_BORDER_WIDTH);
+        const total_width = BUTTON_WIDTH * this.buttons_number + this.title_window_width + 2 * numbers.OUTSIDE_BORDER_WIDTH + 2;
```

Comment:
Nothing wrong about multiplying by 2, but most of the time I prefer `(numbers.OUTSIDE_BORDER_WIDTH << 1)`.

Reply:
I copied that part from `HorizontalMenu.js`. But I will change it.

Reply:
Perfect! Please improve my things 😁